### ISSUE TITLE
emar: Fails with response file argument

### ISSUE THREAD

**Issue by ShadowIce** [emscripten-core/emscripten#8105]
I have a project that I compile with cmake + ninja which fails since 1.38.27 during a call to emar.

```
FAILED: libFoo.a 
cmd.exe /C "cd . && C:\emsdk\emscripten\1.38.27\emar.bat rc libFoo.a  @CMakeFiles\Foo.rsp && cd ."
Traceback (most recent call last):
  File "C:\emsdk\emscripten\1.38.27\emar.py", line 77, in <module>
    sys.exit(run())
  File "C:\emsdk\emscripten\1.38.27\emar.py", line 71, in run
    return shared.run_process(newargs, stdin=sys.stdin).returncode
  File "C:\emsdk\emscripten\1.38.27\tools\shared.py", line 166, in run_process
    proc = Popen(cmd, *args, **kw)
  File "C:\emsdk\python\2.7.13.1_64bit\python-2.7.13.amd64\lib\subprocess.py", line 390, in __init__
    errread, errwrite)
  File "C:\emsdk\python\2.7.13.1_64bit\python-2.7.13.amd64\lib\subprocess.py", line 640, in _execute_child
    startupinfo)
WindowsError: [Error 87] The parameter is incorrect
```
Seems to be related to the changes made in #7945.

**Comment by kripken** [comment]
@sbc100 is fixing more things in that file - does this still fail on current incoming for you?

Can you provide a testcase?

**Comment by ShadowIce** [comment]
I wasn't able to test it with incoming but I created a small example. I was only able to reproduce this on Windows, on Linux everything seems to be working fine.

It looks like it has something to do with the length of the line inside the rsp file. It works if I have fewer files, or shorter paths.

[Example.zip](https://github.com/emscripten-core/emscripten/files/2875833/Example.zip)

**Comment by kripken** [comment]
We do add 8 digits of hashing, which increases filename sizes - perhaps that makes it reach a windows limit?

We could reduce the digits, but only at risk of not solving the problem the hashing is for. Perhaps instead, if this is a windows issue, we could ensure that the total size is not larger than the platform limit (and add less or no hashing, as needed).

**Comment by sbc100** [comment]
The final call to llvm-ar should be don't with a response file, so it should never have a command line that is too long.  I'm assuming this response file is loaded by llvm-ar itself and not interpreted by win32?   This failure looks like it happens before llvm-ar even starts which is why I'm confused.

Can you run with EMCC_DEBUG=1 and see what it says.

**Comment by sbc100** [comment]
Oh wait, this was already fixed in #8049.

**Comment by sbc100** [comment]
This will be fixed once 1.38.28 is released.   Or I think you can `emsdk install emscripten-incoming-xxx` to get a fix today.

**Comment by ShadowIce** [comment]
I'll try it once 1.38.28 is out. emscripten-incoming doesn't compile here because my antivirus detects trojan-like behavior in it (probably false positive) and my IT department doesn't allow me to change anything there.
